company.el

```diff
@@ -1230,7 +1238,12 @@ can retrieve meta-data for them."
                    (list (cons prefix
                                (company--preprocess-candidates candidates))))
              (unwind-protect
-                 (company-idle-begin buf win tick pt)
+                 (if (company-call-backend 'no-cache prefix)
+                     (company-idle-begin buf win tick pt)
+                   (let ((current-prefix (company-call-backend 'prefix))
+                         (ignore-case (company-call-backend 'ignore-case)))
+                     (when (string-prefix-p prefix current-prefix ignore-case)
+                       (company-idle-begin buf win (buffer-chars-modified-tick) (point)))))
```

Comment:
There's a danger here in some cases where you end up on another matching prefix on a different line that is in a different context and wouldn't otherwise match. That could be mitigated by either remembering our line or if `prev` is `async` above, resetting `tick` and `point` in some way that this closure would have access to it. It could also be fine because maybe caching would work that way any way.